src/asmcup/genetics/Gene.java

```diff
@@ -10,16 +12,27 @@ public Gene(byte[] dna, int gen, float score) {
 		this.gen = gen;
 		this.score = score;
 	}
-	
+
+	@Override
 	public int compareTo(Gene other) {
 		float d = score - other.score;
 		
-		if (d == 0) {
+		if (Math.abs(d) < 0.00000001f) {

```

Comment:
Static analysis can say what it wants, but the check here is precisely meant to be "if this equals float zero (positive or negative)". This is (aside from NaN, which isn't handled either way) the only case where `d < 0` and `d > 0` are both false (and consequently where `a > b` and `b > a` could consequently be true, which is what a sort algorithm will complain about.

I disagree with this change.